### PR TITLE
Add NFTManager + tests

### PR DESCRIPTION
**Pull request by tspoff**

### CODE REVIEW

**Comment by tspoff** [comment]
@orenyodfat My JS test file is formatted differently than the rest - is there a contribution / style guide for formatting?

File: contracts/misc/NFTManager.sol [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    function initialize() public initializer {
+        Ownable.initialize(msg.sender);
+    }
+
+    /// @notice Transfer specified NFT to a given recipient
+    /// @param recipient Address to transfer NFT to
+    /// @param nftContract NFT token contract to transfer from (e.g. Cryptokitties contract)
+    /// @param tokenId ID of specific NFT to transfer
+    function sendNFT(address recipient, IERC721 nftContract, uint tokenId) external onlyOwner {
+        nftContract.transferFrom(address(this), recipient, tokenId);
```
Comment (orenyodfat) [inline contracts/misc/NFTManager.sol:17]: should we use safeTransferFrom . see https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/IERC721.sol#L59 .
might even better to use https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/IERC721.sol#L128 as it gives more options to pass data along the call.

**Comment by orenyodfat** [comment]
lets make travis green.

File: contracts/misc/NFTManager.sol (hunk 1 above)
Comment (tspoff) [reply contracts/misc/NFTManager.sol:17]: My issue with safeTransferFrom() is the requirement that receiving contracts must implement IERC721Reciever - there may be a number of contract recipients that don’t implement this.

However, I agree safeTransfer should be the default option unless there’s a specific reason not to use it.

Here's my proposed change, with explicit functions for sending data or no data with safeTransfer(): https://github.com/dOrgTech/arc/pull/1/files

Comment (orenyodfat) [reply contracts/misc/NFTManager.sol:17]: please pr your proposal to this repo.

File: contracts/misc/NFTManager.sol [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    function initialize() public initializer {
+        Ownable.initialize(msg.sender);
+    }
+
+    /// @notice Safe Transfer specified NFT to a given recipient
+    /// @param recipient Address to transfer NFT to
+    /// @param nftContract NFT token contract to transfer from (e.g. Cryptokitties contract)
+    /// @param tokenId ID of specific NFT to transfer
+    function sendNFT(address recipient, IERC721 nftContract, uint tokenId) public onlyOwner {
```
Comment (orenyodfat) [inline contracts/misc/NFTManager.sol:19]: this can be `external`

Comment (orenyodfat) [reply contracts/misc/NFTManager.sol:19]: why this function needed at all ? does sendNFTWithData is not enough ?

File: contracts/misc/NFTManager.sol [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+     * returned can be obtained as `this.onERC721Received.selector`. This
+     * function MAY throw to revert and reject the transfer.
+     * Note: the ERC721 contract address is always the message sender.
+     * @param operator The address which called `safeTransferFrom` function
+     * @param from The address which previously owned the token
+     * @param tokenId The NFT identifier which is being transferred
+     * @param data Additional data with no specified format
+     * @return bytes4 `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`
+     */
+    function onERC721Received(address operator, address from, uint256 tokenId, bytes memory data)
```
Comment (orenyodfat) [inline contracts/misc/NFTManager.sol:57]: please remove unused var name ..
`function onERC721Received(address, address, uint256, bytes )`

File: test/nftmanager.js [hunk 4]
```diff
@@ -82,11 +82,20 @@ const setup = async function(
     account2 = reputationAccount;
   }
   testSetup.proxyAdmin = accounts[5];
+  console.log(
```
Comment (orenyodfat) [inline test/nftmanager.js:85]: why console.log ?

File: test/nftmanager.js [hunk 5]
```diff
@@ -0,0 +1,689 @@
...
+  var testSetup = new helpers.TestSetup();
+  testSetup.standardTokenMock = await ERC20Mock.new(accounts[1], 100);
+  registration = await helpers.registerImplementation();
+  testSetup.reputationArray = [20, 10, 70];
+  var account2;
+  if (reputationAccount === 0) {
+    account2 = accounts[2];
+  } else {
+    account2 = reputationAccount;
+  }
```
Comment (orenyodfat) [inline test/nftmanager.js:83]: this is needed for. something ?

**Pull request closed without merging**